homeassistant/components/netatmo/media_source.py

```diff
@@ -0,0 +1,148 @@
+"""Netatmo Media Source Implementation."""
+import datetime as dt
+import re
+from typing import Optional, Tuple
+
+from homeassistant.components.media_player.errors import BrowseError
+from homeassistant.components.media_source.const import MEDIA_MIME_TYPES
+from homeassistant.components.media_source.error import Unresolvable
+from homeassistant.components.media_source.models import (
+    BrowseMedia,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DATA_CAMERAS, DATA_EVENTS, DOMAIN, MANUFACTURER
+
+MIME_TYPE = "application/x-mpegURL"
+
+
+async def async_get_media_source(hass: HomeAssistant):
+    """Set up Netatmo media source."""
+    return NetatmoSource(hass)
+
+
+class NetatmoSource(MediaSource):
+    """Provide Netatmo camera recordings as media sources."""
+
+    name: str = MANUFACTURER
+
+    def __init__(self, hass: HomeAssistant):
+        """Initialize Netatmo source."""
+        super().__init__(DOMAIN)
+        self.hass = hass
+        self.events = self.hass.data[DOMAIN][DATA_EVENTS]
+
+    async def async_resolve_media(self, item: MediaSourceItem) -> PlayMedia:
+        """Resolve media to a url."""
+        _, location = async_parse_identifier(item)
+        camera_id, event_id = split_path(location)
+        url = self.events[camera_id][event_id].get("media_url")
```

Comment:
You use `.get` here, so it could be `None`. If you know the key always exist, use square brackets.
```suggestion
        url = self.events[camera_id][event_id].get["media_url"]
```